src/domain/payments/index.types.d.ts

```diff
@@ -1,11 +1,27 @@
-type PriceRatio = {
+type PriceRatio<S extends WalletCurrency> = {
+  convertFromOther(other: bigint): PaymentAmount<S>
+  convertFromWallet(btcWalletAmount: PaymentAmount<S>): bigint
+  convertFromWalletToFloor(btcWalletAmount: PaymentAmount<S>): bigint
+  convertFromWalletToCeil(btcWalletAmount: PaymentAmount<S>): bigint
+  otherUnitPerWalletUnit(): DisplayCurrencyBasePerSat
+}
+
+type WalletPriceRatio = {
   convertFromUsd(convert: UsdPaymentAmount): BtcPaymentAmount
   convertFromBtc(convert: BtcPaymentAmount): UsdPaymentAmount
   convertFromBtcToFloor(convert: BtcPaymentAmount): UsdPaymentAmount
   convertFromBtcToCeil(convert: BtcPaymentAmount): UsdPaymentAmount
   usdPerSat(): DisplayCurrencyBasePerSat
 }
 
+type DisplayPriceRatio<S extends WalletCurrency, T extends DisplayCurrency> = {
```

Comment:
New object type coming in here that outputs `DisplayCurrencyObect<T>` types

Reply:
maybe is better to modify DisplayAmount

Reply:
Yup, but this would be a transition step while we have the two separate types in our codebase still. I haven't changed any of the application code as yet to use this new type. Just establishing the pattern first and looking for feedback on that as our eventual target.

Reply:
https://github.com/GaloyMoney/galoy/pull/2334/files#r1108850976